### DISCUSSION TITLE
getstaticprops not working with _app.js

### DISCUSSION THREAD

**Discussion by DanSallau** [vercel/next.js#12439] [Help]
I have my simple application that use to work with getInitialProps, following the recommendations on the doc page, i changed it getStaticProps and getStaticPaths. The problem now is, while bot the two functions get executed, the new props returned by getServerSideProps, getStaticProps doesn't get passed unto my page/component. I have a _app.js with little implementation of getInitialProps like below

```
class MyApp extends App<MyAppProps, {}> {
  static async getInitialProps({ Component, ctx }) {
    return {
      pageProps: Component.getInitialProps
        ? await Component.getInitialProps(ctx)
        : {},
    };
  }

  render() {
    const { Component, pageProps, router, store } = this.props;
    const persistor = persistStore(store);
    return (
         //some providers etc.
                <Component router={router} {...pageProps} />
  )
}

}
    
```
The above use to work for me and the new props from getInitialProps get passed. However, it doesn't work with getStaticProps and getServerSideProps, Please how do i enhance my implementation to allow passing of such props ? Any hep will be appreciated.

**Comment by DeepGoel** [comment]
I am also facing the same issue. Props for the page is not getting passing when using the getInitialProps in my _app.tsx file.

**Comment by brunocrosier** [comment]
I think this will help. Not sure that `getStaticProps` is available for `_app.js` at this moment in time..

https://github.com/zeit/next.js/discussions/10949#discussioncomment-1203
